lib/mocha_runner.coffee

```diff
@@ -0,0 +1,19 @@
+ArgumentBuilder = require './argument_builder'
+mochaExecutablePath = require.resolve('mocha/bin/mocha')
+{spawn} = require 'child_process'
+
+
+class MochaRunner
+
+  constructor: ({argv, cwd}) ->
+    @argumentBuilder = new ArgumentBuilder {argv, cwd}
+
+
+  run: (done) ->
+    @argumentBuilder.build (err, args) ->
+      if err then return done err
+      childProcess = spawn mochaExecutablePath, args, stdio: 'inherit'
```

Comment:
I would try to use the Mocha JS API here, instead of forking a subprocess: https://github.com/mochajs/mocha/blob/master/test/jsapi/index.js